src/executors/mod.rs

```diff
@@ -63,20 +64,55 @@ pub async fn execute_http_function(request: &Request, function: FunctionInfo) ->
             pyo3_asyncio::tokio::into_future(function_output?)
         })?
         .await?;
-        let mut res: HashMap<String, String> =
-            Python::with_gil(|py| -> PyResult<HashMap<String, String>> { output.extract(py) })?;
+
+        // convert the output to a PyDict
+        let res = Python::with_gil(|py| -> Result<Response> {
+            let output: Py<PyDict> = output.extract(py)?;
```

Comment:
converting it to a response shouldn't be crazy hard tbh. I will try again tomorrow 😅

Reply:
I tried implementing it since `FromPyObject` conflicts with the `Clone` trait. The implementation becomes very hairy.